src/GitHub.App/Services/RepositoryService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using GitHub.Api;
+using GitHub.Extensions;
+using GitHub.Primitives;
+using Octokit.GraphQL;
+using static Octokit.GraphQL.Variable;
+
+namespace GitHub.Services
+{
+    [Export(typeof(IRepositoryService))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class RepositoryService : IRepositoryService
+    {
+        static ICompiledQuery<string> readParentOwnerLogin;
+        readonly IGraphQLClientFactory graphqlFactory;
+
+        [ImportingConstructor]
+        public RepositoryService(IGraphQLClientFactory graphqlFactory)
+        {
+            Guard.ArgumentNotNull(graphqlFactory, nameof(graphqlFactory));
+
+            this.graphqlFactory = graphqlFactory;
+        }
+
+        public async Task<string> ReadParentOwnerLogin(HostAddress address, string owner, string name)
```

Comment:
I wonder if this would be more useful as a general `FindParent` method that returns a `(string owner, string repositoryName)`? A parent repository might not have the same name as its child and we may want to support this as some point.

Reply:
Ah yeah, good point. I will make this change.

Reply:
I changed this to return the repository name and owner. We don't actually use the `name` yet,: `UriString` doesn't support getting a new URL with a different name as well as owner so want to do that in a separate PR, this is big enough as it is ;)